WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewView.swift

```diff
@@ -169,10 +169,9 @@ private extension ProductDetailPreviewView {
                           isFocused: focusedField == .name,
                           shouldEnableUndo: viewModel.hasChangesToProductName,
                           onUndoEdits: {
-            // TODO: update this
-            viewModel.productName = viewModel.generatedProduct?.name ?? ""
+            viewModel.undoEdits(in: ProductDetailPreviewViewModel.ProductDetailField.name)
         })
-        .focused($focusedField, equals: FocusedField.name)
+        .focused($focusedField, equals: ProductDetailPreviewViewModel.ProductDetailField.name)
```

Comment:
Nit:

```suggestion
        .focused($focusedField, equals: .name)
```

Similar updates can be made in the other places below for brevity.

Reply:
Done in fb9019c11a9c0fada795e74bb5db2a1f7bf8cf33